fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/IndexingScrubDangling.java

```diff
@@ -184,33 +186,58 @@ private CompletableFuture<Boolean> scrubIndexRangeOnly(@Nonnull FDBRecordStore s
-                store.addRecordReadConflict(indexEntry.getPrimaryKey());
-                store.getContext().ensureActive().clear(keyBytes);
+        final IndexingCommon.IndexContext indexContext = common.getIndexContext();
+        final IndexEntry indexEntry = indexEntryResult.get();
+        if (indexContext.isSynthetic) {
+            return store.loadSyntheticRecord(indexEntry.getPrimaryKey()).thenApply(syntheticRecord -> {
+                if (syntheticRecord.getConstituents().isEmpty()) {
+                    // None of the constituents of this synthetic type are present, so it must be dangling
+                    List<Tuple> primaryKeysForConflict = new ArrayList<>(indexEntry.getPrimaryKey().size() - 1);
+                    SyntheticRecordType<?> syntheticRecordType = store.getRecordMetaData().getSyntheticRecordTypeFromRecordTypeKey(indexEntry.getPrimaryKey().get(0));
+                    for (int i = 0; i < syntheticRecordType.getConstituents().size(); i++) {
+                        if (!(syntheticRecordType.getConstituents().get(i).getRecordType() instanceof NestedRecordType) && indexEntry.getPrimaryKey().get(i + 1) != null) {
+                            primaryKeysForConflict.add(indexEntry.getPrimaryKey().getNestedTuple(i + 1));
+                        }
+                    }
+                    scrubDanglingEntry(store, indexEntry, primaryKeysForConflict);
+                }
+                return null;
+            });
+        } else {
+            return store.loadIndexEntryRecord(indexEntry, IndexOrphanBehavior.RETURN).thenApply(indexedRecord -> {
+                if (!indexedRecord.hasStoredRecord()) {
+                    // Here: Oh, No! this index is dangling!
+                    scrubDanglingEntry(store, indexEntry, List.of(indexEntry.getPrimaryKey()));
+                }
+                return null;
+            });
+        }
+    }
+
+    private void scrubDanglingEntry(@Nonnull FDBRecordStore store, @Nonnull IndexEntry indexEntry, @Nonnull List<Tuple> conflictPrimaryKeys) {
+        danglingCount.incrementAndGet();
+        timerIncrement(FDBStoreTimer.Counts.INDEX_SCRUBBER_DANGLING_ENTRIES);
+        final Tuple valueKey = indexEntry.getKey();
+        final byte[] keyBytes = store.indexSubspace(common.getIndex()).pack(valueKey);
+
+        if (LOGGER.isWarnEnabled() && logWarningCounter > 0) {
+            logWarningCounter--;
+            LOGGER.warn(KeyValueLogMessage.build("Scrubber: dangling index entry",
+                            LogMessageKeys.KEY, valueKey,
+                            LogMessageKeys.PRIMARY_KEY, indexEntry.getPrimaryKey())
+                    .addKeysAndValues(common.indexLogMessageKeyValues())
+                    .toString());
+        }
+        if (scrubbingPolicy.allowRepair()) {
+            // remove this index entry
+            // Note that there no record can be added to the conflict list, so we'll add the index entry itself.
```

Comment:
This comment is a little garbled and I can't actually work out exactly what it wants to say.

Reply:
@MMcM , I'll be happy to fix this comment in a follow up PR that I'm planning.